### PR TITLE
Pictures backup custom date

### LINKED ISSUE #284: Choise a date for sync picture

### PR DESCRIPTION
**Pull request by LunarX**
Closes #284

### CODE REVIEW

File: app/src/main/res/layout/view_date_input.xml [hunk 1]
```diff
@@ -26,7 +26,6 @@
         style="@style/TextInputLayout.OutlinedBox"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
```
Comment (KevinBoulongne) [inline app/src/main/res/layout/view_date_input.xml:28]:
```suggestion
        android:layout_height="wrap_content"
        android:layout_marginTop="@dimen/marginStandardMedium"
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt [hunk 2]
```diff
@@ -57,6 +57,7 @@ class SyncSettingsActivity : BaseActivity() {
     private val selectDriveViewModel: SelectDriveViewModel by viewModels()
     private var oldSyncSettings: SyncSettings? = null
     private var editNumber = 0
+//    private var customDate = Date().startOfTheDay()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt:60]:
```suggestion
```

File: app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt [hunk 3]
```diff
@@ -100,6 +101,15 @@ class SyncSettingsActivity : BaseActivity() {
             }
         }
 
+        syncSettingsViewModel.customDate.observe(this) {
+            if (it == null) {
+                syncDatePicker.isGone = true
+            } else {
+                syncDatePicker.text = it.format(FORMAT_DATE_CLEAR_MONTH)
+                syncDatePicker.isVisible = true
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt:109]:
```suggestion
                syncDatePicker.apply {
                    text = it.format(FORMAT_DATE_CLEAR_MONTH)
                    isVisible = true
                }
```

File: app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt [hunk 4]
```diff
@@ -279,13 +278,17 @@ class SyncSettingsActivity : BaseActivity() {
     private fun saveSettings() {
         saveButton.showProgress()
         lifecycleScope.launch(Dispatchers.IO) {
-            val saveOldPictures = syncSettingsViewModel.saveOldPictures.value == true
+            val date = when (syncSettingsViewModel.saveOldPictures.value!!) {
+                SyncSettings.SavePicturesDate.SINCE_NOW -> Date()
+                SyncSettings.SavePicturesDate.SINCE_FOREVER -> Date(0)
+                SyncSettings.SavePicturesDate.SINCE_DATE -> syncSettingsViewModel.customDate.value ?: Date()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt:284]: _Alt+Enter / "Add import for"_

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt [hunk 5]
```diff
@@ -304,9 +308,32 @@ class SyncSettingsActivity : BaseActivity() {
...
+            .setTheme(R.style.MaterialCalendarThemeBackground)
+            .setSelection(
+                // Need to account for time zones as MaterialDatePicker expects GMT+0. Else setting the picker to anything between
+                // 02.01.22 00:00:00 and 02.01.22 00:59:59 results in the date picker being set to 01.01.22 instead when the device uses GMT+1
+                dateToSet.time + TimeZone.getDefault().getOffset(dateToSet.time)
+            )
+            .setCalendarConstraints(calendarConstraints)
+            .build().apply {
+                addOnPositiveButtonClickListener { syncSettingsViewModel.customDate.value = Date(it).startOfTheDay() }
+                show(supportFragmentManager, "syncDate")
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt:329]: Wrong tag fragment

File: app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt [hunk 6]
```diff
@@ -193,22 +208,7 @@ class SyncSettingsActivity : BaseActivity() {
...
-                .setTitle(getString(R.string.syncSettingsButtonSaveDate))
-                .setSingleChoiceItems(items, checkedItem) { _, which ->
-                    syncOldMedia = which == 1
-                }
-                .setPositiveButton(R.string.buttonConfirm) { _, _ ->
-                    syncSettingsViewModel.saveOldPictures.value = syncOldMedia
-                }
-                .setNegativeButton(R.string.buttonCancel) { _, _ -> }
-                .setCancelable(false).show()
+            SelectSaveDateBottomSheetDialog().show(supportFragmentManager, "SyncSettingsSaveDateBottomSheetDialog")
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt:211]: Wrong tag fragment

File: app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt [hunk 7]
```diff
@@ -100,6 +100,17 @@ class SyncSettingsActivity : BaseActivity() {
...
         }
 
+        syncSettingsViewModel.customDate.observe(this) { date ->
+            syncDatePicker.apply {
+                if (date == null) {
+                    isGone = true
+                } else {
+                    isVisible = true
+                    text = date.format(FORMAT_DATE_CLEAR_MONTH)
+                }
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/drive/ui/menu/settings/SyncSettingsActivity.kt:110]:
```suggestion
                isGone = date == null
                text = date?.format(FORMAT_DATE_CLEAR_MONTH) ?: ""
```

**Review by JorisBodin — CHANGES_REQUESTED** [review]

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**